### ISSUE TITLE
Unbundle oemof and solph

### ISSUE THREAD

**Issue by uvchik** [oemof/oemof-solph#639]
Unbundle `oemof` and `solph` to clarify the structure.

At the moment we find `oemof=solph` in the public discussion. In my opinion `oemof` is `solph`, `tespy`, `feedinlib`, `windpowerlib`, `cydets`, `oemof_thermal` and `oemof_tabular`.

Therefore, I would suggest to create a `solph` repository. The `oemof` repository would be a dummy package that installs all stable (typical?) oemof packages.

Open question:
1. Where do we want to place the network stuff?
    a. as subpackage in the solph package
    b. in a new package (oemof_base, ...)
    c. ...
2. Will the `outputlib` part of the solution above (1) or an own package?
3 ...

I just want to start an open discussion about that and we may substantiate it on the next dev-meeting.

**Comment by p-snft** [comment]
In my opinion, unbundling makes perfectly sense. I agree, that the oemof repository can be used as a meta-repository that pulls all oemof packages. This way, also backward compatibility can probably be maintained. Further, it would make things more comfortable, as e.g. the examples might be automatically updated. (We could use [git submodules](https://git.wiki.kernel.org/index.php/GitSubmoduleTutorial) for that.

Concerning the questions, I have the feeling that both, network and outputlib, could be seperate repositories:
1. Network can be used without solph. (Potentially. I did not yet do much for [scoop](https://github.com/oemof/oemof/commits/features/scoop), yet.)
2. Outputlib can potentially work without solph. In fact, here at ENaQ we detach optimisation from analysation. The second step relies on a model containing results, which might e.g. have been post-processed.

Then:
3. Would e.g. the TESPy package become oemof.tespy?

**Comment by simnh** [comment]
I like the idea of unbundling, it helps also with other 'subpackages' extra packages that are now located in their own repositories like oemof tabular and oemof-thermal. 

I like the idea to be able to install all repos by themselves but also get a oemof bundle where you might maybe also select which packages to include. I do not know how that could be possible in a good usability way. 

However I have to mention, that there will not be much time from my side to do the unbundling etc.

**Comment by p-snft** [comment]
One further thought: From a technical perspective, feedinlib and demandlib are very similar. They both prepare/ create time series. Does it make sense to merge them?

Background: We created some scripts to align data read in from overlapping time series present in various csv files that might also have different time zones or even time formats. Currently, I am not sure where to put them, but they would definitely fit into some `oemof.timeserieslib`.

**Comment by joroeder** [comment]
> One further thought: From a technical perspective, feedinlib and demandlib are very similar. They both prepare/ create time series. Does it make sense to merge them?

How about putting all other precalcaultions of timesieres there as well? like heatpump COPs?

**Comment by uvchik** [comment]
It is time to start the unbundling process.

1. We will rename the oemof repository to oemof-solph!
2. We will rename the oemof.organisation to oemof (protest still possible).
3. We already created oemof-tools and oemof-network (renaming still possible).
4. There will be merge conflicts for the existing PRs in oemof, but I will help to solve them. Be aware that the source directory is named `src` and the `doc` directory is named `docs` because. The `src` dir helps us to integrate the packages in one namespace. We renamed docs because this is typical in python packages.
5. All these packages can be used like this: 
```python
from oemof import solph
from oemof import tools
from oemof import network
...
```
- network contains base packages for the description of systems as a bipartite graph.
- solph is the model generator
- tools is a small package containing useful functions

The integration in the oemof namespace works fine and in my opinion every package can decide whether it wants to  be integrated or not (to be discussed).
